test/integration/formservice/formservice.js

```diff
@@ -0,0 +1,155 @@
+      .should('have.attr', 'src', '/formservice/1/1/1');
+
+    cy
+      .window()
+      .then(win => {
+        win.postMessage({ message: 'form:pdf', args: {
+          definition: {
+            'components': [
+              {
+                'key': 'patient.fields.insurance',
+                'type': 'container',
+                'input': true,
+                'label': 'Insurance',
+                'tableView': false,
+                'components': [
+                  {
+                    'key': 'name',
+                    'type': 'textfield',
+                    'input': true,
+                    'label': 'Insurance Name',
+                    'tableView': true,
+                  },
+                ],
+              },
+              {
+                'key': 'patient.fields.pcp',
+                'type': 'container',
+                'input': true,
+                'label': 'PCP',
+                'tableView': false,
+                'components': [
+                  {
+                    'key': 'name',
+                    'type': 'textfield',
+                    'input': true,
+                    'label': 'PCP Name',
+                    'tableView': true,
+                  },
+                ],
+              },
+              {
+                'key': 'patient.fields.address',
+                'type': 'container',
+                'input': true,
+                'label': 'Address',
+                'tableView': false,
+                'components': [
+                  {
+                    'key': 'state',
+                    'type': 'textfield',
+                    'input': true,
+                    'label': 'State',
+                    'tableView': true,
+                  },
+                ],
+              },
+            ],
+          },
+          formData: {
+            'patient': {
+              'fields': {
+                'insurance': {
+                  'name': 'Test Insurance Name',
+                },
+                'pcp': {
+                  'name': 'Test PCP Name',
+                },
+              },
+            },
+          },
+          formSubmission: {
+            'patient': {
+              'fields': {
+                'pcp': {
+                  'name': 'Test PCP Name',
+                },
+                'address': {
+                  'state': 'Test State Name',
+                },
+                'insurance': {
+                  'name': 'Test Insurance Name',
+                },
+              },
+            },
+          },
+          contextScripts: {},
+          reducers: [],
+        } }, win.origin);
+      });
+
+    cy
+      .get('[name="data[patient.fields.insurance][name]"]')
+      .should('have.value', 'Test Insurance Name');
+
+    cy
+      .get('[name="data[patient.fields.pcp][name]"]')
+      .should('have.value', 'Test PCP Name');
+
+    cy
+      .get('[name="data[patient.fields.address][state]"]')
+      .should('have.value', 'Test State Name');
+  });
```

Comment:
Do we need this test form to have all three of these input fields?

We could just trim it down to just the one input field and the test would serve the same purpose, right? Would be cleaner and less lines of code.

Reply:
Just went with reducing it to one input.